pkg/login/login.go

```diff
@@ -0,0 +1,83 @@
+package login
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/sirupsen/logrus"
+)
+
+// Client is able to detect available login providers and login to DC/OS.
+type Client struct {
+	http   *httpclient.Client
+	logger *logrus.Logger
+}
+
+// NewClient creates a new login client.
+func NewClient(baseClient *httpclient.Client, logger *logrus.Logger) *Client {
+	return &Client{
+		http:   baseClient,
+		logger: logger,
+	}
+}
+
+// Providers returns the supported login providers for a given DC/OS cluster.
+func (c *Client) Providers() (map[string]*Provider, error) {
+	resp, err := c.http.Get("/acs/api/v1/auth/providers")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	providers := make(map[string]*Provider)
+	if resp.StatusCode == 200 {
+		err := json.NewDecoder(resp.Body).Decode(&providers)
+		if err == nil {
```

Comment:
What is `err != nil`? We are already in the case of a status code 200 thus we should not expect one at this point.

Reply:
This is indeed a code path which shouldn't happen, unless something is broken in DC/OS, or the API endpoint introduced some breaking changes. However in the unlikely case where it happens I'd rather fail with an explicit error rather than silencing it and having the CLI failing/misbehaving later on.

Reply:
> However in the unlikely case where it happens I'd rather fail with an explicit error rather than silencing it and having the CLI failing/misbehaving later on.
But that is the case at the moment, no? I would expect a:
```
		if err == nil {
			return providers, nil
		} else {
			return nil, err
		}
```

Reply:
Ah yes indeed, I was confused, updated.